File: 121. Best Time to Buy and Sell Stock/121. Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        if(prices.length == 0) return 0;
        
        int min = prices[0], max = 0, sum = 0;
        for(int i = 1; i < prices.length; i++){
            sum = prices[i] - min;
            min = Math.min(min, prices[i]);
            max = Math.max(max, sum);
        }
        return max;
    }
}

File: 121. Best Time to Buy and Sell Stock/121. Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  if (prices.length == 0) return 0;

  let min = prices[0],
    max = 0,
    sum = 0;

  for (let i = 1; i < prices.length; i++) {
    sum = prices[i] - min;
    min = Math.min(min, prices[i]);
    max = Math.max(sum, max);
  }
  return max;
};
